### PR TITLE
Feature/synchronize

### PR DESCRIPTION
**Pull request by xieyxclack**
To synchronize the code with our gitlab

### CODE REVIEW

File: federatedscope/attack/example_attack_config/reconstruct_IG_fedavg_opt_on_femnist.yaml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 use_gpu: True
 device: 0
+<<<<<<< HEAD
```
Comment (joneswong) [inline federatedscope/attack/example_attack_config/reconstruct_IG_fedavg_opt_on_femnist.yaml:3]: both modified

File: federatedscope/attack/example_attack_config/reconstruct_fedavg_opt_on_femnist.yaml [hunk 2]
```diff
@@ -1,6 +1,11 @@
 use_gpu: True
 device: 0
+<<<<<<< HEAD
```
Comment (joneswong) [inline federatedscope/attack/example_attack_config/reconstruct_fedavg_opt_on_femnist.yaml:3]: both modified

**Review by joneswong — CHANGES_REQUESTED** [review]
It seems that there are still several conflicts.

**Comment by xieyxclack** [comment]
> It seems that there are still several conflicts.

I have resolved the conflicts accordingly.

**Comment by rayrayraykk** [comment]
How about remove `docker_file/feratedscope_jupyterhub`?

**Review by joneswong — APPROVED** [review]
Approve!

**Pull request merged**